data/render/src/lib.rs

```diff
@@ -21,6 +23,64 @@ pub fn validate(tags: &Tags, entries: &Vec<Entry>) -> Result<(), Box<dyn Error>>
     Ok(())
 }
 
+#[tokio::main]
+pub async fn check_deprecated(token: std::string::String, entries: &mut Vec<Entry>) -> Result<Vec<Entry>, Box<dyn Error>> {
+    let github = Github::new(
+        String::from("user-agent-name"),
+        Credentials::Token(token),
+    )?;
+
+    let mut entries_tmp: Vec<Entry> = entries.to_vec();
+    for entry in &mut entries_tmp {
+        if entry.source.is_some() {
```

Comment:
I personally prefer a bit more flat structure.

You can restructure this code to negate if condition here and put "continue" in it. That way you'll avoid multiple indentations.

This applies also to the if conditions that you put below.